File: languages/kotlin/src/model/Table.kt
package model

import box.UserInfo
import java.io.BufferedReader
import java.io.File

class Table(fileName: String) {

    // Get e Set de fileName é gerado pelo construtor
    // Get de userInfoList é gerado manualmente
    val userInfoList: List<UserInfo> = this.deserializeFile(fileName)

    /**
     * Método privado para encriptar senhas
     * password: String -> senha a ser encriptada
     * RETURN: String -> nova senha
     */
    private fun deserializeFile(fileName: String): List<UserInfo> {

        var list = mutableListOf<UserInfo>()

        val bufferedReader: BufferedReader = File(fileName).bufferedReader()
        var header: Boolean = true

        bufferedReader.useLines {

            lines -> lines.forEach {

                if(header) {
                    header = false
                }else {

                    val values = it.split(",")

                    val user: String = values[0].trim()
                    val password: String = values[1].trim()
                    val credit: Double = values[2].trim().toDouble()

                    val userInfo = UserInfo(user, password, credit)

                    list.add(userInfo)
                }
            }
        }

        bufferedReader.close()
        return list.toList()
    }
}

File: languages/R/src/model/Table.R
# Criando uma classe "Table"
Table = 
  R6Class("Table",
          # Definindo valores e funções públicas
          public = list(
            
            # Construtor da classe
            # fileName -> nome do arquivo
            initialize = function(fileName) {
              private$fileName = fileName
              private$userInfoList = private$deserializeFile(fileName)
            },

            #Método getter de "fileName"  
            getFileName = function() { return(private$fileName) },
            
            #Método getter de "userInfoList"
            getUserInfoList = function() { return(private$userInfoList) }
          ),

          # Definindo valores e funções privadas      
          private = list(
            fileName = NULL,
            userInfoList = c(),
            
            # Método privado para conversão do arquivo .csv em uma lista de "UserInfo"
            # fileName -> Nome do arquivo
            # RETURN -> Lista convertida de "UserInfo"
            deserializeFile = function(fileName) {
              # Obs.: Leitura de arquivo extremamente rápida (própria para big data)
              dt = fread(fileName)
              vec = 1:dim(dt)[1]
              
              # Lendo Binário de UserInfo
              loadcmp("./bin/UserInfo.Rc")
              
              # Obs.: Conversão demorada
              list = sapply(vec, function(index){
                user = dt$user[index]
                password = dt$password[index]
                credit = dt$credit[index]
                userInfo = UserInfo$new(user, password, credit)
                return(userInfo)
              })
              
              return(list)
            }
          )
  )